src/server/gc_worker/gc_manager.rs

```diff
@@ -514,14 +514,18 @@ impl<S: GcSafePointProvider, R: RegionInfoProvider + 'static, E: KvEngine> GcMan
             self.check_if_need_rewind(&progress, &mut need_rewind, &mut end);
 
             let controller: Arc<(Mutex<usize>, Condvar)> = Arc::clone(&task_controller);
-            let cb = Box::new(move |_res| {
-                let (lock, cvar) = &*controller;
-                let mut current_tasks = lock.lock().unwrap();
-                *current_tasks -= 1;
-                cvar.notify_one();
-                AUTO_GC_PROCESSED_REGIONS_GAUGE_VEC
-                    .with_label_values(&[PROCESS_TYPE_GC])
-                    .inc();
+            let cb = Box::new(move |res| {
+                if let Err(Error(box ErrorInner::GcWorkerTooBusy)) = res {
+                    // the task is actually not scheduled, so no action here.
```

Comment:
It looks still not a nice-looking solution to me 🤔 Can we consider replacing it with some other synchronization methods instead of using mutex and condvar?

Reply:
The current approach is simpler and easier to pick for the LTS version. Alternatively, an atomic variable can be used to avoid mutex, with periodic checks in `gc_a_round`.

Further refactoring could be left as a TODO task for GC restructuring. We could leave a TODO here.